### PR TITLE
feat: simpler entity resolvers in dynamic mocks

### PR DESCRIPTION
**Pull request by obmarg**
I kinda had something like this in mind when I was writing the dynamic mocks, but I didn't need them so didn't want to spend the time.

Since #2427 has started using entity resolvers now seems as good a time as any to revisit.

### CODE REVIEW

File: crates/graphql-mocks/src/dynamic/entity_resolvers.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            repr,
+        }
+    }
+
+    pub fn add_error(&self, error: ServerError) {
+        self.inner_context.query_env.errors.lock().unwrap().push(error);
+    }
+
+    /// Returns the key of the entity we need to resolve
+    pub fn key(&self) -> serde_json::Value {
```
Comment (Finistere) [inline crates/graphql-mocks/src/dynamic/entity_resolvers.rs:25]: Would name it `representation`, it also includes fields from `@requires` not just from `@key`

**Review by Finistere — APPROVED** [review]

File: crates/graphql-mocks/src/dynamic/entity_resolvers.rs (hunk 1 above)
Comment (obmarg) [reply crates/graphql-mocks/src/dynamic/entity_resolvers.rs:25]: hmm, fair point.  Although the representation usually also includes the __typename....

Comment (obmarg) [reply crates/graphql-mocks/src/dynamic/entity_resolvers.rs:25]: Tempted to just not offer this as a convenience

Comment (Finistere) [reply crates/graphql-mocks/src/dynamic/entity_resolvers.rs:25]: ah didn't realize it, could just remove it yeah

Comment (obmarg) [reply crates/graphql-mocks/src/dynamic/entity_resolvers.rs:25]: yeah, just going to bin this.  Could implement something that looks at the actual keys defined for the entity, but not worth it unless we need it.

**Pull request merged**